c/parallel/src/jit_templates/CMakeLists.txt

```diff
@@ -0,0 +1,25 @@
+file(GLOB_RECURSE template_files "*.h")
+set(jit_template "${CMAKE_CURRENT_BINARY_DIR}/jit_template.h")
+set(jit_template_src "${CMAKE_CURRENT_BINARY_DIR}/jit_template.cpp")
+set(cpp_entry "${CMAKE_CURRENT_LIST_DIR}/jit_entry.h")
+
+file(MAKE_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}")
+
+add_custom_command(
+    OUTPUT "${jit_template}"
```

Comment:
You need to capture `template_pre.h.in` and `template_post.h.in` in your dependecy set.

Reply:
Really you should use `DEPFILE` and be passing the makefile fragment from the compiler to get the full dependency set ( see -MT / -M )